src/topic_based_system.cpp

```diff
@@ -38,6 +38,32 @@
 #include <rclcpp/executors.hpp>
 #include <topic_based_ros2_control/topic_based_system.hpp>
 
+namespace
+{
+/** @brief Sums the total rotation for joint states that wrap from 2*pi to -2*pi
+when rotating in the positive direction */
+void sumRotationFromMinus2PiTo2Pi(const double current_wrapped_rad, const double previous_wrapped_rad,
+                                  double& total_rotation)
+{
+  double delta = current_wrapped_rad - previous_wrapped_rad;
+
+  // Check for discontinuities due to the jump from 2*pi to -2*pi and correct them
+  if (delta > M_PI)
+  {
+    // The change is large and positive, but it should be small and negative
+    delta -= 4.0 * M_PI;
+  }
+  else if (delta < -M_PI)
+  {
+    // The change is large and negative, but it should be small and positive
+    delta += 4.0 * M_PI;
+  }
+
+  // Add the corrected delta to the total rotation
+  total_rotation += delta;
+}
+}  // namespace
+
```

Comment:
Should we use one of the functions from https://github.com/ros/angles/blob/ros2/angles/include/angles/angles.h?

Reply:
Thanks for the tip on using that library!